### PR TITLE
Adds support for Terraform 0.13 and above

### PR DESCRIPTION
**Pull request by danvaida**
Upcoming release tag: `v3.0.0`

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 output "transit_gateway_vpc_attachment_id" {
   description = "Identifier of the Transit Gateway VPC Attachment"
-  value       = element(concat(aws_ec2_transit_gateway_vpc_attachment.this.*.id, list("")), 0)
+  value       = element(concat(aws_ec2_transit_gateway_vpc_attachment.this.*.id, tolist([])), 0)
```
Comment (SerikDM) [inline outputs.tf:3]: Looks like value changes `[""]` -> `[]`. But then you do `concat` and pick first element anyway. It shouldn't matter.
Out of curiosity, why is this done? looks like it's equal:
`value = aws_ec2_transit_gateway_vpc_attachment.this.*.id`

**Review by SerikDM — APPROVED** [review]

File: outputs.tf (hunk 1 above)
Comment (danvaida) [reply outputs.tf:3]: I believe this was done like that as a result of using the `terraform 0.13upgrade` command and then simply following the suggestion. I must admit that I haven't looked at it up close. Will test it now and if what you showed is equivalent, I'll go with it.

Comment (danvaida) [reply outputs.tf:3]: Circling back to this, we can see that just like all the other resources in this module, the [TGW VPC attachment resource](https://github.com/35up/terraform-aws-transit-gateway-satellite/blob/master/main.tf#L4) is also acting based on the `create` toggle. Should that toggle be false, the resource wouldn't be created and this output would thrown an error, hence the so-called splat operator is used instead.

Comment (SerikDM) [reply outputs.tf:3]: Does it work like this if TGW VPC is not created? My concern is if it's not created, then `concat` will return empty list(while previously it returned list with empty string `[""]`), then `element` will try to get first element of empty list.. and I couldn't find anything about this situation in docs :unamused:

**Pull request merged**